pkg/api/logs.go

```diff
@@ -0,0 +1,54 @@
+package api
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+func (h handler) getAPIV1NamespaceResourceLog(w http.ResponseWriter, r *http.Request) {
+	log.Println("called getAPIV1NamespaceResourceLog")
+
+	namespace := mux.Vars(r)["namespace"]
+	resource := mux.Vars(r)["resource"]
+	name := mux.Vars(r)["name"]
+	container := r.URL.Query().Get("container")
+	previous, _ := strconv.ParseBool(r.URL.Query().Get("previous"))
+
+	logFileName := fmt.Sprintf("%s.log", container)
+	if previous {
+		logFileName = fmt.Sprintf("%s-previous.log", container)
+	}
+
+	fileName := filepath.Join(h.clusterData.ClusterResourcesDir, resource, "logs", namespace, name, logFileName)
+	data, err := ioutil.ReadFile(fileName)
+	if err != nil {
+		log.Println("failed to load file", err)
+		if os.IsNotExist(err) {
+			// try reading from -logs-errors.log file
+			errFileName := filepath.Join(h.clusterData.ClusterResourcesDir, resource, "logs", namespace, name, fmt.Sprintf("%s-logs-errors.log", container))
+			data, err = ioutil.ReadFile(errFileName)
```

Comment:
technically this is not a pod log but errors that occurred while collecting logs.  this could get confusing if someone misinterprets this output

Reply:
if a pod is not in running/terminated state(eg: `ContainerCreating`) and we run `kubectl logs` we will get the error
```
➜  ~ k logs busybox12
failed to get log stream: container "kotsadm-postgres" in pod "kotsadm-postgres-0" is waiting to start: ContainerCreating
```
If we run collector on the above pod, the result will be saved into `-logs-errors.log` 
Since we are trying to simulate what `kubectl` does, It would be nicer to show the error message.

```
➜  ~ k logs kotsadm-postgres-0
failed to get log stream: container "kotsadm-postgres" in pod "kotsadm-postgres-0" is waiting to start: ContainerCreating
```

the other errors in `-logs-errors.log`  can be anything related to running collectors, time-outs,

Since the `sbctl` users know it is running on support-bundle, I hope returning these errors instead of 500 is more meaningful.